src/components/AppPagination.vue

```diff
@@ -0,0 +1,57 @@
+<template>
+    <div class="app-pagination">
+      <v-pagination
+        v-model="currentPage"
+        :length="totalPages"
+        total-visible="10"
+        @previous="previousPage"
+        @next="nextPage"
+        color="success"
+      ></v-pagination>
+    </div>
+</template>
+
+<script>
+export default {
+  name: 'AppPagination',
+  props: {
+    actionType: {
+      type: String
+    }
+  },
+  computed: {
+    currentPage: {
+      get: function () {
+        return this.$store.state.currentPage
+      },
+      set: function (value) {
+        this.$store.commit('setPage', value)
+        this.$store.dispatch(this.actionType, this.currentPage)
+      }
+    },
+    totalPages () {
+      return this.$store.state.totalPages <= 500 ? this.$store.state.totalPages : 500
```

Comment:
and what about pages 501+ ? Users will never know about them, right?
just return `this.$store.state.totalPages`, but total visible could be 6 or 7

Reply:
Done